core/modules/widgets/widget.js

```diff
@@ -557,6 +557,29 @@ Widget.prototype.allowActionPropagation = function() {
 	return true;
 };
 
+Widget.prototype.generateTransclusionFootprint = function() {
+	var parentTransclusionWidget = this.findParentTransclusionWidget(),
+	    node = this,
+	    footprint = node.parentWidget.children.indexOf(node);
+	while(node = node.parentWidget) {
+		if(node === parentTransclusionWidget) {
+			break;
+		} else if(node.parentWidget && node.parentWidget.children) {
+			footprint = footprint + "-" + node.parentWidget.children.indexOf(node);
+		}
+	}
+	return footprint;
+};
+
+Widget.prototype.findParentTransclusionWidget = function() {
+	var node = this;
+	var transclusionVariable = this.getVariable("transclusion");
+	while(node.getVariable("transclusion") === transclusionVariable) {
```

Comment:
Variables that are undefined within a widget are passed to the parent widget for resolution. Thus, when you're walking up a stack of ancestors, the first time you get the chosen transclusion value won't be the widget in which it is defined, but rather the deepest descendent widget that doesn't redefine that value.

Reply:
ah, I see. so it works by accident I guess ... I need to think this through, haven't thought about that yet

Reply:
@Jermolene - here I'm walking up the stack of ancestors until the `parentTransclusionWidget` is reached. that's the one that defines the transclusion value (?) ... the while loop is for detecting all ancestor widgets up to the `parentTransclusionWidget` where the transclusion variable is the same

Reply:
... hm, does all this mean that I'm one widget off?

Reply:
hm... does this mean that I'm one widget off?

Reply:
@Jermolene, I think this works fine, if by accident or not I don't know :smile: